File: practical05/fibonacci.c
#include<stdio.h>
#include<stdlib.h>

//Function declaration/prototye
//input argument: Fn-1 and Fn-2
//on exit: Fn and Fn-1
void fibonacci(int *a, int *b);

int main(){

  int n,i;
  int f0=0, f1=1;
  printf("Ebter a positive integer\n");
  scanf("%d",&n);
 
  if(n<1){
    printf("The number is not positive.\n");
    exit(1);
  }

  printf("The fibonacci sequence is: \n");
  printf("%d, %d, ",f0,f1);
  
  //Calculating fibonacci sequence from 2
  for(i=2;i<=n;i++){
    fibonacci(&f1,&f0);
    printf("%d, ", f1);
    if((i+1)%10 == 0) printf("\n");

  }
  return 0;
}

void fibonacci(int *a, int *b){
  int next;
  //*a=Fn-1, *b=Fn-2, next=Fn
  next = *a+*b;
  *b=*a;
  *a=next;
}

File: practical05/fibonacci.f90
module fib
  contains
  subroutine fibonacci(a,b)
    integer (kind=4), intent(inout) :: a,b
    integer (kind=4) :: next
   
    !a=fn-1 b=fn-2 next=fn
    !on exit: a=fn, b=fn-1
    next=a+b
    b=a
    a=next
  end subroutine fibonacci
end module fib

program fibonacciseries
  use fib
  implicit none

  integer (kind=4) :: n, i, f0=0, f1=1
  !Enter a number
  write(6, *), 'Enter a positive integer'
  read(5, *), n
  !check
  if(n .lt. i) then
    write(6, *), 'The number is not positive'
    stop
  endif

  write(6, *), 'The fibonacci seqwence is: '
  write(6, '(i0, a, i0, a)', advance='NO')

  do i =2,n
    call fibonacci(f1,f0);
    write(6, '(i0,a)', advance='NO'), f1, ', '
  end do
end program fibonacciseries
